### PR TITLE
Add service worker

### LINKED ISSUE #20: Request for guides to be deployed for mobility
@raiseadead @QuincyLarson @Bouncey 

A discussion in CT started about deployment for mobility of FCC resources. "Guides" was suggested as target.

This post opens a discussion about the desirability and feasibility of this request and if decided to start working on a road map.

### PR DESCRIPTION
**Pull request by Bouncey**
closes #20

### CODE REVIEW

**Comment by Bouncey** [comment]
I have committed the wrong set of favicons, I'll amend the PR soon

File: src/layouts/components/search/SearchBar.jsx [hunk 1]
```diff
@@ -72,21 +75,33 @@ class SearchBar extends PureComponent {
...
+        type='text'
+        value={ searchTerm }
+      /> :
+      <FormControl
+        className='input'
+        disabled={ true }
+        onChange={ this.handleChange }
+        placeholder='&#xf002; It looks like you are currently offline'
+        type='text'
+        value={ searchTerm }
```
[the remark this answers is not in the document]
Comment (Bouncey) [reply src/layouts/components/search/SearchBar.jsx]: Hmm, I didn't think of that case

File: src/layouts/redux/index.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { createTypes } from 'redux-create-types';
+import { createAction, handleActions } from 'redux-actions';
+
+const initialState = {
+  isOnline: true
```
[the remark this answers is not in the document]
Comment (Bouncey) [reply src/layouts/redux/index.js]: A ternary default should cover browsers without the navgator. Something like
```js
{
  isOnline: (navigator && 'onLine' in navigator) ? navigator.onLine : true;
}
```

**Comment by systimotic** [comment]
And of course Timo messes up the merge conflict resolution _twice_ the one time he thinks it's an easy one and is able to use GitHub's built-in editor to do it...
Sorry about that, should be resolved now.

File: static/manifest.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (Bouncey) [inline static/manifest.json]: @systimotic here is the `manifest.json`. All the gatsby plugin does it generate this file. You still have to manually input the config into the plugin options

File: src/head/favicons.js [hunk 4]
```diff
@@ -1,232 +1,32 @@
...
-    <meta
-      content='/'
-      name='msapplication-TileImage'
-    />
+  <link
+    href='/apple-touch-icon.png'
+    rel='apple-touch-icon'
+    sizes='180x180'
+  />,
+  <link
```
[the remark this answers is not in the document]
Comment (Bouncey) [reply src/head/favicons.js]: updated

File: src/layouts/components/search/SearchBar.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Bouncey) [reply src/layouts/components/search/SearchBar.jsx]: This case is handled :+1:

**Comment by Bouncey** [comment]
`sw.js` is only created on build

**Comment by QuincyLarson** [comment]
@systimotic Could you go ahead and merge this once you think it's ready?

**Comment by Bouncey** [comment]
@systimotic I have addressed merge conflicts with this one

**Comment by QuincyLarson** [comment]
@systimotic @Bouncey How is this coming along? Is this ready to QA and merge? Or are there still outstanding Review issues (I noticed it hasn't yet been dismissed).

**Comment by Bouncey** [comment]
@QuincyLarson @systimotic this is ok to go on my end

**Comment by systimotic** [comment]
@Bouncey Some good things, some issues 😞
1. New favicons are 👌, manifest works well too
2. The `browserconfig.xml` is located in `static/assets/favicons/` and not referenced with a meta tag. It can not be found.
3. I can't test on desktop Chrome because of the callstack error
4. When adding to the homescreen on Android Chrome, it looks amazing, apart from the square corners on the icon. On the first load, it didn't load the index content properly, but I can't reproduce it anymore. (being able to add this to my homescreen is really cool)
5. Updating the online status no longer works. I can't confidently say why, but I'm guessing it has something to do with there being two `updateOnlineStatus` functions.
6. Learning that the service worker was only generated on build was really helpful. I can confirm that it works, but it has some issues. It's showing a few warnings on build, and I'm seeing the result of those warnings when I try to test loading without an internet connection. The good news: I get some sort of a response! The bad news: not everything is loading. The warnings were about files being too large to cache. I can tell that Bootstrap and FontAwesome aren't loading, a lot of the content won't display, and some scripts are having trouble too. Not seeing the favicon anymore either.
Build warnings
Firefox errors in Dutch (translation: "laden" = "loading", "mislukt" = "failed")

**Comment by QuincyLarson** [comment]
@Bouncey Were you able to take a look at @systimotic's requested changes and resolve all of them?

**Comment by Bouncey** [comment]
Closeing as way behind and it would be easier to start again

**Pull request closed without merging**